### ISSUE TITLE
Request: Allow precision to be chosen for played values

### ISSUE THREAD

**Issue by Jedsters** [FrSkyRC/ETHOS-Feedback-Community#3491]
Currently a value can be taken from a VAR, pot, slider etc. and played via a SF. This could be a value such as 27.3% 
I'm guessing that very few will be interested in the .3% so please could we have an option to either allow us to specify what precision the value should be played or possibly whether decimals should just be ingored.

I guess there are various options here, a precision could be specified as 
0.1% i.e. play as per now
0.5% i.e. play to nearest 0.5% so 27.5% in the above example
1% to play 27%
5% to play 25% 
or maybe anything in-between?
etc. etc.

I suspect this could just be done at a radio level or possibly model level to apply to all played values. 
Personally I'd just be happy with a tick box as to whether decimal values should be played tx wide,  but others may have differing opinions.

One day, no hurry.

Thanks as always!

**Comment by Toffs** [comment]
Agreed, most of the standard programming options, curves, mixes don't go into 0.1% increment precision anywhere else that I know of > an option to give that precision would be fine, but default rounding to nearest whole integer would be better IMO.


**As part of this '0.1% is too small an increment / not helpful' in Vars:**
I struggle with Repurpose Trims or using a custom trim mix in Vars because the adjust value is always tiny 0.1% increments. I don’t understand why this is, and need to create a second Vars that multiplies the first Vars by x10 in order to use a custom trim mix.
How do people use a trim mix in a Vars that isn't affect by 2000 increments (+/-100.0%) , rather than 200?

**Comment by Jedsters** [comment]
For clarity, my request was more to only have interger values played in SF, so if a VAR was e.g. 27.3% the value played would be 27% (the .3% adds no value, for me anyway, and takes extra time potentially delaying the next play).

However, I do similarly wonder as to whether decimals are needed in VARs, but I'll let others decide on that, they could possibly have a use with multiply or divide? 
I did have a play and used a VAR -100 to 100 as a weight into an output channel to see what difference 0.1% makes and it appears that 0.2% is required to change the output channel by 1us - but I know that's not an exact value, I'm guessing 0.1% is actually 0.512us or similar.

**Comment by Toffs** [comment]
Ahh, got it.  Not having 0.1% increments (or option to have whole integers) will also fix the decimal played issue :)

**Comment by doppleganger53** [comment]
On the flipside, more precision is sometimes necessary, so configurable voice precision is ideal. For example, reading out voltages at the cell level, there's a significant difference between 3.65V and 3.7V, currently everything is rounded.

**Comment by Jedsters** [comment]
Very good point, voltages are an example of somewhere decimals are needed, I guess similar could be other telemetry. So maybe it's a case of having the option for percentages and VARs being rounded rather than across the board for everything.

**Comment by tmcallahan** [comment]
Please add me to the list of people who would like to have more precision or adjustable precision. I thought I was doing something wrong, so I checked here to see if it's a known issue.  When Ethos read it out the voltage and it rounds up or down to one decimal point,  it is not precise enough to know whether I need to land asap because it says 3.7,   or if it is really 3.75 and I have another 30 seconds or so.  It's a ball park reading, but it should be more precise for voltage..

**Comment by bob01** [comment]
++ for me please - use case system.playNumber(cellVoltage, UNIT_VOLTS, 2) for 3.75 etc - as previous comment noted there is a huge difference between 3.75 and 3.70v

**Comment by Didier-L** [comment]
Same issue in Ethos with Play Value function. See #4117
Did not get an answer to know if it will be implemented (or not) in a future release.

**Comment by MiRe-CZ** [comment]
Hello @bsongis-frsky,

can be please this enhancement in the system.playNumber() implemented? It is really annoying to hear incorrect rounded numbers, especially when using announcements during flight for tuning of some parameter crucial for exact course measurement

Thank you

BR Milan

**Comment by doppleganger53** [comment]
Any hope of seeing this in 1.7.0? Voice callouts for lipo cell voltage are dangerously inaccurate when trying to get the most out of a battery.

**Comment by tmcallahan** [comment]
@bsongis-frsky Any hope or news on this feature/enhancement? Its been bumped for 2 years, and would be really appreciated to prioritize this in an upcoming release.

**Comment by MiRe-CZ** [comment]
@bsongis-frsky: I guess limiting announcements to two decimal digits would be good approach? I do not see any case with need for higher precision (but I can be mistaken). In such case why not tell both digits together? Excluding cn, for all rest languages there are wavs ready for numbers from 10 to 99, so only wavs for 01 to 09 are missing. Still not very easy, but I suppose it would satisfy almost all users

**Comment by Jedsters** [comment]
> [@bsongis-frsky](https://github.com/bsongis-frsky): I guess limiting announcements to two decimal digits would be good approach? I do not see any case with need for higher precision (but I can be mistaken). In such case why not tell both digits together? Excluding cn, for all rest languages there are wavs ready for numbers from 10 to 99, so only wavs for 01 to 09 are missing. Still not very easy, but I suppose it would satisfy almost all users

I'm not sure what you mean when you say tell both digits together. If you're suggesting decimals could be announced, e.g.  10.25 could be announced as 'ten point twenty-five' I would **not** be in favour of that.

**Comment by MiRe-CZ** [comment]
Exactly that I am suggesting with aim to have some reasonably simple and acceptable solution asap. Otherwise, well, this ticket is quite old and not really in the focus, apparently implementation of full scope (e.g. telemetry allows 4-digits precision ...) would be difficult. Honestly I am personally interested to have the system.playNumber function correctly running, with the parameter decimals = 2. I also think that 'ten point twenty-five' is much easier to understand than 'ten point two five'

**Comment by bsongis** [comment]
I didn't say I wouldn't work on it, just that it is not a simple fix.

**Comment by Jedsters** [comment]
> I also think that 'ten point twenty-five' is much easier to understand than 'ten point two five'

I'm afraid that's just a matter of opinion, one which I disagree with, if it said 'ten point twenty-five' I'd end up having to do the mental conversion to ten point two five in my head. 
Please no!

**Comment by bsongis** [comment]
I suppose that in english there is a "standard" way to speak the decimals? I will check that point with @lthole 

In french I am qualified to decide what I will do :)

Other languages ... I have to contact all translators one by one

**Comment by Wolepo** [comment]
In German, the ones are combined first and then the tens—for example, "point one-and-twenty" or "point nine-and-ninety."

**Comment by Dsrenger** [comment]
What????

In German 12,33 is "twelve point three three"

I am an mechanical engineer and never heard somebody saying "twelve point thirty three"

**Comment by lthole** [comment]
In English 12 point three three also.

**Comment by mawzthefinn** [comment]
Yes, standard English is each digit spoken individually after the decimal. 

American might differ ;-)

**Comment by bsongis** [comment]
ah no!!!

**Comment by MiRe-CZ** [comment]
Decimals are grouped in the standard and casual Czech, so not spoken individually. But I will leave @janekx to confirm officially

**Comment by tmcallahan** [comment]
Im not a scientist, but in america, we generally would say 3.75 as three point Seven five. This would be my preference, but at the end of the day, I can handle either way, as long as it communicates in some way the 3.75 in full. I'm very appriciative that this is finally getting some attention.

**Comment by Wolepo** [comment]
The comma was translated as a period.
So, "comma-three-twenty." = ,23
Or simply use the System/11.wav through 99.wav files
